src/file.ts

```diff
@@ -2595,6 +2601,63 @@ class File extends ServiceObject<File> {
     });
   }
 
+  isPublic(): Promise<IsPublicResponse>;
+  isPublic(callback: IsPublicCallback): void;
+  /**
+   * @callback IsPublicCallback
+   * @param {?Error} err Request error, if any.
+   * @param {boolean} resp Whether file is public or not.
+   */
+  /**
+   * @typedef {array} IsPublicResponse
+   * @property {boolean} 0 Whether file is public or not.
+   */
+  /**
+   * Check whether this file is public or not.
+   * @param {IsPublicCallback} [callback] Callback function.
+   * @returns {Promise<IsPublicResponse>}
+   *
+   * @example
+   * const {Storage} = require('@google-cloud/storage');
+   * const storage = new Storage();
+   * const myBucket = storage.bucket('my-bucket');
+   *
+   * const file = myBucket.file('my-file');
+   *
+   * //-
+   * // Check whether the file is publicly accessible.
+   * //-
+   * file.isPublic(function(err, resp) {
+   *   if (err) {
+   *     console.error(err);
+   *     return;
+   *   }
+   *   console.log(`the file ${file.id} is public: ${resp}`) ;
+   * })
+   * //-
+   * // If the callback is omitted, we'll return a Promise.
+   * //-
+   * file.isPublic().then(function(data) {
+   *   const resp = data[0];
+   * });
+   */
+
+  isPublic(callback?: IsPublicCallback): Promise<IsPublicResponse> | void {
+    gaxiosRequest({
+      method: 'HEAD',
+      url: `http://${this.bucket.name}.storage.googleapis.com/${this.id}`,
```

Comment:
I think we need to encode the name:

```js
url: `http://${this.bucket.name}.storage.googleapis.com/${encodeURIComponent(this.name)}`,
```

Reply:
sure, done

It only matters in the case where `file.name` property is changed on an already existing `file` object, which would render all other file properties' values incorrect in respect to the new `file.name`.
Shouldn't the `file.name` property be made `readonly`?

Reply:
> It only matters in the case where file.name property is changed

Do you mean the `name` property is already encoded? I didn't see that in the code.

> Shouldn't the file.name property be made readonly?

I suppose. `name` wasn't intended to be a public property at all.